session.cpp

```diff
@@ -25,26 +30,27 @@ int session::handle_request(const boost::system::error_code& error,
     size_t bytes_transferred){
 
   std::vector<char>message_request = convert_buffer();
+
   HttpRequest request(message_request);
+  http_response response;
 
-  std::string url = request.getUrl();
+  std::string url = request.getUrl();  
   std::string function = get_function_from_url(url);
 
-  std::cout << "Function: " << function << std::endl;
+  if (!error) {
 
-  if (!error){
-    if (function == "echo_dir")
-      handle_write(error, bytes_transferred);
-    else if (function == "static_dir"){
-      //stuff to handle static file handling goes here. 
-      //we should modularize our code so there are separate classes 
-      //for sending static files and for sending echos
-      std::cout << "Static File Handling Yet To Be Implemented" << std::endl;
+    if (function == "echo_dir") {
+      handle_echo(error, bytes_transferred, response);
+    } else if (function == "static_dir") {
+      handle_static(url, response);
     }
```

Comment:
I think they want these to be separate classes? 

"The request handlers for echoing requests and serving files should use a common interface. This means they should both inherit from the same base class, defined in a separate .h file." 

That is from the spec. So I think handle_static and handle_echo should be in separate classes that inherit from some common handler class.